### PR TITLE
Refactor Dependencies::process() with exceptions

### PR DESCRIPTION
**Pull request by siwinski**

### CODE REVIEW

File: src/Dependencies.php [hunk 1]
```diff
@@ -94,26 +94,30 @@ private function __construct()
...
+                $loadedDependency = null;
 
                 foreach ($dependency as $firstExistsDependency) {
-                    if (file_exists($firstExistsDependency)) {
-                        $dependencyToLoad = $firstExistsDependency;
+                    try {
+                        requireFile($firstExistsDependency);
+                        $loadedDependency = $firstExistsDependency;
                         break;
+                    } catch (\RuntimeException $e) {
```
Comment (siwinski) [inline src/Dependencies.php:112]: For the change here, I was trying to make this block of code use the same conditionals logic as `Fedora\Autoloader::requireFile()` without copying-and-pasting the conditionals.  This does have the overhead of an Exception object creation (from `Fedora\Autoloader::requireFile()`) and the catch though.  Perhaps a copy-and-paste of the conditionals here would be better or a common function (something like `Fedora\Autoloader::fileLoadable()`?) that this code block and `Fedora\Autoloader::requireFile()` could both use?

File: src/Dependencies.php [hunk 2]
```diff
@@ -94,26 +94,30 @@ private function __construct()
...
+     *
+     * @throws \RuntimeException If required and no file is successfully loaded
+     *     via {@link requireFile()}.
      */
     protected static function process(array $dependencies, $required)
     {
         foreach ($dependencies as $dependency) {
             if (is_array($dependency)) {
-                $dependencyToLoad = null;
+                $loadedDependency = null;
```
Comment (remicollet) [inline src/Dependencies.php]: A boolean can be enough ?

File: src/Dependencies.php [hunk 3]
```diff
@@ -94,26 +94,30 @@ private function __construct()
...
 
-                if ($required || isset($dependencyToLoad)) {
-                    if (!isset($dependencyToLoad)) {
-                        $dependencyToLoad = end($dependency);
-                    }
-
-                    requireFile($dependencyToLoad);
+                if ($required && empty($loadedDependency)) {
+                    throw new \RuntimeException(sprintf(
+                      'Files not found: "%s"',
```
Comment (remicollet) [inline src/Dependencies.php]: for consistency with other exception use single quote around each name.

**Review by remicollet — APPROVED** [review]
no problem for me, feel free to merge for 0.2.0 is you want

File: src/Dependencies.php (hunk 2 above)
Comment (siwinski) [reply src/Dependencies.php]: Yes!  Updated.

File: src/Dependencies.php (hunk 3 above)
Comment (siwinski) [reply src/Dependencies.php]: Updated.

**Comment by siwinski** [comment]
Updated

**Pull request merged**